src/coreclr/vm/gcinfodecoder.cpp

```diff
@@ -1810,7 +1810,7 @@ bool GcInfoDecoder::IsScratchRegister(int regNum,  PREGDISPLAY pRD)
 {
     _ASSERTE(regNum >= 0 && regNum <= 31);
 
-    return (regNum <= 21 && regNum >= 4);
+    return (regNum <= 21 && ((regNum >= 4) || (regNum == 1)));
```

Comment:
Why would one store GC pointers in the return address register?

Reply:
because the ra may be used as a scratch register at some causes.